tests/host_callback_test.py

```diff
@@ -915,20 +912,50 @@ def func(y):
     assertMultiLineStrippedEqual(self, """
 { lambda  ; a.
   let b c = id_tap[ arg_treedef=PyTreeDef(tuple, [*,*])
-                    batch_dims=(None, 0)
                     func=_print
-                    transforms=('batch',) ] 3.00 a
+                    transforms=(('batch', (None, 0)),) ] 3.00 a
       d = add c 3.00
   in (d,) }""", str(api.make_jaxpr(vmap_func)(vargs)))
     with hcb.outfeed_receiver():
       res_vmap = vmap_func(vargs)
     assertMultiLineStrippedEqual(self, """
-batch_dims: (None, 0) transforms: ('batch',)
+transforms: (('batch', (None, 0)),)
 [ 3.00
   [4.00 5.00] ]
    """, testing_stream.output)
     testing_stream.reset()
 
+  def test_double_vmap(self):
+    # A 2D tensor with x[i, j] = i + j using 2 vmap
+    def sum(x, y):
+      return hcb.id_print(x + y, output_stream=testing_stream)
+    def sum_rows(xv, y):
+      return api.vmap(sum, in_axes=(0, None))(xv, y)
+    def sum_all(xv, yv):
+      return api.vmap(sum_rows, in_axes=(None, 0))(xv, yv)
+
+    xv = jnp.arange(5, dtype=np.int32)
+    yv = jnp.arange(3, dtype=np.int32)
+    assertMultiLineStrippedEqual(self, """
+{ lambda  ; a b.
+  let c = broadcast_in_dim[ broadcast_dimensions=(1,)
+                            shape=(3, 5) ] a
+      d = reshape[ dimensions=None
+                   new_sizes=(3, 1) ] b
+      e = add c d
+      f = id_tap[ arg_treedef=*
+                  func=_print
+                  transforms=(('batch', (0,)), ('batch', (0,))) ] e
+  in (f,) }""", str(api.make_jaxpr(sum_all)(xv, yv)))
```

Comment:
`textwrap.dedent` is a nice way to handle multi-line string with correct indentation:
```
  Yes:
  import textwrap

  long_string = textwrap.dedent("""\
      This is also fine, because textwrap.dedent()
      will collapse common leading spaces in each line.""")
```
http://google.github.io/styleguide/pyguide.html#310-strings

Reply:
The `assertMultiLineStrippedEqual` already strips multiple spaces, so the problem is not that one. The current indentation comes out of how I update those strings: copy and paste from a test failure message. I guess I could indent the whole thing afterwards, but I did not bothered doing so. Do you think it is important? I am afraid that adding more cost to updating will make this too costly (it may already be so).

Reply:
I don't think it's too important, though it does make the tests slightly easier to wrote. My editor makes it easy to add indentation to a block of code all at once but it does indeed require a few more keystrokes.